control-plane/csi-driver/controller/src/pvwatcher.rs

```diff
@@ -0,0 +1,143 @@
+    }
+
+    pub(crate) async fn start_watcher(&self) -> anyhow::Result<()> {
+        info!("Starting PV Garbage Collector");
+        let cloned_self = self.clone();
+        tokio::spawn(async move {
+            cloned_self.handle_missed_events().await;
+        });
+        watcher(self.pv_handle.clone(), ListParams::default())
+            .touched_objects()
+            .try_for_each(|pvol| async {
+                self.process_object(pvol).await;
+                Ok(())
+            })
+            .await
+            .unwrap();
+        Ok(())
+    }
+
+    async fn process_object(&self, pv: PersistentVolume) {
+        let mut deletion_candidate = false;
+        let mut our_pv = false;
+        match pv.metadata.clone().deletion_timestamp {
+            Some(_) => deletion_candidate = true,
+            None => {}
+        }
+
+        if let Some(provisioner) = &pv.spec.as_ref().unwrap().csi {
+            if provisioner.driver == CSI_PLUGIN_NAME {
+                our_pv = true;
+            }
+        }
+
+        if deletion_candidate && our_pv {
+            if let Some(reclaim_policy) =
+                &pv.spec.as_ref().unwrap().persistent_volume_reclaim_policy
+            {
+                if let Some(phase) = &pv.status.as_ref().unwrap().phase {
+                    if reclaim_policy == "Retain" && phase == "Released" {
+                        debug!(
+                            "PV :{} is a deletion candidate",
+                            &pv.metadata.clone().name.unwrap()
+                        );
+                        if let Some(provisioner) = &pv.spec.as_ref().unwrap().csi {
+                            delete_volume(&provisioner.volume_handle.to_string()).await;
+                        }
+                    }
+                    if phase == "Bound" {
+                        match self
+                            .pv_handle
+                            .get_opt(&pv.metadata.clone().name.unwrap())
+                            .await
+                        {
+                            Ok(pvol) => match pvol {
+                                Some(_) => debug!(
+                                    "PV {} present on API server",
+                                    &pv.metadata.clone().name.unwrap()
+                                ),
+                                None => {
+                                    debug!(
+                                        "PV :{} is a deletion candidate",
+                                        &pv.metadata.clone().name.unwrap()
+                                    );
+                                    if let Some(provisioner) = &pv.spec.as_ref().unwrap().csi {
+                                        delete_volume(&provisioner.volume_handle.to_string()).await;
+                                    }
+                                }
+                            },
+                            Err(e) => error!("Error while verifying if PV is present: {:?}", e),
```

Comment:
can we add error as a key, example:
`Err(error) => error!(error, "Error while verifying if PV is present"),`